src/lib/downloadReachability.js

```diff
@@ -0,0 +1,93 @@
+const DEFAULT_DOWNLOAD_HOST_PREFIX = 'https://dl.';
+const DOWNLOAD_NOTIFICATION_SHOWN_KEY = 'download_host_notification_shown';
+const DOWNLOAD_HOST_FALLBACK_KEY = 'download_host_fallback';
+const DEFAULT_HOST_REGEX = /^https:\/\/dl\d+\./;
+
+/**
+ * Checks if the url is a download host, but not the default download host.
+ *
+ * @public
+ * @param {string} downloadUrl - Content download URL, may either be a template or an actual URL
+ * @return {boolean} - HTTP response
+ */
+export function isCustomDownloadHost(downloadUrl) {
+    return (
+        downloadUrl && !downloadUrl.startsWith(DEFAULT_DOWNLOAD_HOST_PREFIX) && !!downloadUrl.match(DEFAULT_HOST_REGEX)
+    );
+}
```

Comment:
Does your code work properly on dev with dl.jpress.inside-box.net and dl.app.jpress.inside-box.net? I don't think it's super important for the fallback logic to work on dev since there's only one download host, but please make sure we can't get stuck into some edge case.

Reply:
Yes this works on dev since I only check the DL, but this will have to change due to your comment below.